src/services/BackendService/BackendService.ts

```diff
@@ -56,7 +58,19 @@ export class BackendService implements BackendInterface {
     const region = 'CA';
     const exposureConfigurationUrl = `${this.retrieveUrl}/exposure-configuration/${region}.json`;
     captureMessage('getExposureConfiguration', {exposureConfigurationUrl});
-    return (await fetch(exposureConfigurationUrl, FETCH_HEADERS)).json();
+    const timeoutSecs = Number(Config.DOWNLOAD_TIMEOUT_SECONDS);
+
+    const abortController = new AbortController();
+    const opts = Object.assign(FETCH_HEADERS, {signal: abortController.signal});
+
+    if (timeoutSecs > 0) {
+      clearTimeout(this.configFetchTimeout);
+      this.configFetchTimeout = setTimeout(() => {
```

Comment:
setTimeout for too long in js layer is not recommend. We should move this to native layer then.